### DISCUSSION TITLE
Redis scaler with non-ASCII chars

### DISCUSSION THREAD

**Discussion by pappacena** [kedacore/keda#4590] [Q&A / Need Help] [answered]
I've been using KEDA to control a Celery+Redis autoscaling. Basically, autoscaling based on Redis lists.

Now that I'm using priority queues, some of those lists [includes non-ascii characters](https://docs.celeryq.dev/en/latest/userguide/routing.html#redis-message-priorities) (there is a [bug in Celery](https://github.com/celery/celery/issues/4028#issuecomment-1565062816) that prevents me from changing the separator), and it seems that the autoscaler doesn't work well in this case, with the HPA configuration generating errors like this:

```
unable to get external metric mynamespace/s0-redis-<my-metric-name-with-non-ascii-char>/&LabelSelector{MatchLabels:map[string]string{scaledobject. ││ keda.sh/name: xxx,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get s0-redis-xxxx ││ mon-<my-metric-name-with-non-ascii-char>external.metrics.k8s.io)
```

I wonder if it would be possible to manually configure the external metric name to avoid special characters, like adding a `metricName` to redis scaler metadata config, like [Prometheus scaler has](https://github.com/kedacore/keda/blob/d917094cadbb5fa132e2530c70a5ddfd94710471/pkg/scalers/prometheus_scaler.go#L243).

Maybe something similar to this: https://github.com/kedacore/keda/compare/main...pappacena:keda:redis-allow-metric-name. I'm just unsure because I saw that the `metricName` parameter is deprecated in the current version of Prometheus (not sure why exactly it got deprecated, though).

Any thoughts? If this sounds useful, I can prepare a PR to include the `metricName` parameter.

**Comment by JorTurFer** [comment, marked as answer]
Hi @pappacena 
We have deprecated the `metricName` on all the scaler where they aren't part of the scaler requirements itself. But definitivelly, KEDA should sanitize the metric names to prevent these cases. 
Are you willing to open a PR sanitizing the metric names before responding it?

**Reply by pappacena** [reply]
@JorTurFer sure! I'll try to separate some time during this week to work on a PR for that. 
Thanks for your help!

**Comment by Alekky09** [comment]
Is this fixed, or will it get fixed soon? I'm running into the same exact issue.

**Reply by JorTurFer** [reply]
What KEDA version are you using? From memory, I don't think so, but I'd recommend not using `metricName` anymore because v2.12 will remove it from scalers where isn't something related with the scaling itself (like in redis scalers) after the deprecation period. So if the fix isn't released yet, it won't be released

**Reply by Alekky09** [reply]
Yeah, I guessed so. I think the fix for this should rather be on the Celery side, using non-ascii characters as separators in queue names baffles me.
Anyway, I'm using the 2.4.0 ver.

**Reply by JorTurFer** [reply]
oh, I thought that you were using `metricName` property, I have just understood that your issue is related with the list name itself 🤔 
I think that we are sanitizing the name, but we could ensure that list name doesn't have non-ASCII chars. But in any case, the fix won't be backported to v2.4 because it's a version with almost 2 years. Is there any reason for not upgrading KEDA?
